yt_dlp/extractor/youtube/_base.py

```diff
@@ -595,6 +600,31 @@ def _get_sid_authorization_header(self, origin='https://www.youtube.com', user_s
 
         return ' '.join(authorizations)
 
+    @functools.cached_property
+    def is_authenticated(self):
+        return bool(self._has_auth_cookies())
+
+    def _has_auth_cookies(self):
+        yt_sapisid, yt_1psapisid, yt_3psapisid = self._get_sid_cookies()
+        # YouTube doesn't appear to clear 3PSAPISID when rotating cookies (as of 2025-04-26)
+        # But LOGIN_INFO is cleared and should exist if logged in
+        has_login_info = 'LOGIN_INFO' in self._youtube_cookies()
+        return has_login_info and (yt_sapisid or yt_1psapisid or yt_3psapisid)
+
+    def _request_webpage(self, *args, **kwargs):
+        response = super()._request_webpage(*args, **kwargs)
+
+        # Check that we are still logged-in and cookies have not rotated after every request
+        if self._had_cookie_auth and not self._has_auth_cookies():
+            self._had_cookie_auth = False
```

Comment:
since `is_authenticated` is cached, can't we just use that instead of adding `_had_cookie_auth`?

```suggestion
        if self.is_authenticated and not self._has_auth_cookies():
```

Reply:
technically is_authenticated is a catch-all for all auth types and not necessarily for cookies. but we can use it for now..